exec/jointLklDM.cc

```diff
@@ -714,17 +624,29 @@ int main(int argc,char* argv[])
 		  // if file exists, read histo and use it
 		  else
 		    cout << "Ok!" << endl;	   	      
-	        }
+		}
 	      
 	      // Set the units of g for the different samples (this also creates the fHdNdEpSignal histo
 	      if(isDecay)
-	        fullLkl->SetDMDecayUnitsForG(mass);
+		fullLkl->SetDMDecayUnitsForG(mass);
 	      else	      
-	        fullLkl->SetDMAnnihilationUnitsForG(mass);
+		fullLkl->SetDMAnnihilationUnitsForG(mass);
+	      
+	      // in case simulations were requested with DM cont, do them now
+	      if(isSimulation && mcsv>0 && mcmass>0)
+		{		  
+		  cout << "  * Simulating observations for <sv> = " << mcsv << " and m_DM = " << mcmass << endl;
+		  
```

Comment:
For my understanding, this is where we will need to build the histo for mcmass, mcChannel and mcProcess and pass it on SImulateDataSamples, isn't?